src/decorators.ts

```diff
@@ -194,7 +214,13 @@ function _query(
  * @param target A single element by id or EventTarget to target
  */
 export const listen = (eventName: string, target: string|EventTarget) =>
-    (proto: any, methodName: string) => {
+    (proto: ElementPrototype, methodName: string) => {
+      if (!proto.constructor._addDeclarativeEventListener) {
+        console.error(
```

Comment:
I think it'd be better to throw than warn here.

Reply:
Done.